osu.Game.Rulesets.Rush/Replays/RushReplayFrame.cs

```diff
@@ -41,49 +41,51 @@ public void FromLegacy(LegacyReplayFrame currentFrame, IBeatmap beatmap, ReplayF
             if (currentFrame.MouseRight2) Actions.Add(RushAction.GroundSecondary);
 
             if (currentFrame.MouseX != null)
-                Actions.AddRange(getActionsFromFlags((RushSideActionFlags)currentFrame.MouseX));
+                Actions.AddRange(getActionsFromFlags((RushActionFlags)currentFrame.MouseX));
         }
 
-        public LegacyReplayFrame ToLegacy(IBeatmap beatmap)
-        {
-            ReplayButtonState state = ReplayButtonState.None;
-
-            if (Actions.Contains(RushAction.AirPrimary)) state |= ReplayButtonState.Left1;
-            if (Actions.Contains(RushAction.AirSecondary)) state |= ReplayButtonState.Left2;
-            if (Actions.Contains(RushAction.GroundPrimary)) state |= ReplayButtonState.Right1;
-            if (Actions.Contains(RushAction.GroundSecondary)) state |= ReplayButtonState.Right2;
-
-            return new LegacyReplayFrame(Time, (float)getFlagsFromActions(Actions), 0f, state);
-        }
+        public LegacyReplayFrame ToLegacy(IBeatmap beatmap) => new LegacyReplayFrame(Time, (float)getFlagsFromActions(Actions), 0f, ReplayButtonState.None);
 
-        private static RushSideActionFlags getFlagsFromActions(IEnumerable<RushAction> actions)
+        private static RushActionFlags getFlagsFromActions(IEnumerable<RushAction> actions)
         {
-            RushSideActionFlags flags = RushSideActionFlags.None;
+            RushActionFlags flags = RushActionFlags.None;
 
-            if (actions.Contains(RushAction.AirTertiary)) flags |= RushSideActionFlags.AirTertiary;
-            if (actions.Contains(RushAction.AirQuaternary)) flags |= RushSideActionFlags.AirQuaternary;
-            if (actions.Contains(RushAction.GroundTertiary)) flags |= RushSideActionFlags.GroundTertiary;
-            if (actions.Contains(RushAction.GroundQuaternary)) flags |= RushSideActionFlags.GroundQuaternary;
+            if (actions.Contains(RushAction.AirPrimary)) flags |= RushActionFlags.AirPrimary;
+            if (actions.Contains(RushAction.AirSecondary)) flags |= RushActionFlags.AirSecondary;
+            if (actions.Contains(RushAction.AirTertiary)) flags |= RushActionFlags.AirTertiary;
+            if (actions.Contains(RushAction.AirQuaternary)) flags |= RushActionFlags.AirQuaternary;
+            if (actions.Contains(RushAction.GroundPrimary)) flags |= RushActionFlags.GroundPrimary;
+            if (actions.Contains(RushAction.GroundSecondary)) flags |= RushActionFlags.GroundSecondary;
+            if (actions.Contains(RushAction.GroundTertiary)) flags |= RushActionFlags.GroundTertiary;
+            if (actions.Contains(RushAction.GroundQuaternary)) flags |= RushActionFlags.GroundQuaternary;
 
             return flags;
         }
 
-        private static IEnumerable<RushAction> getActionsFromFlags(RushSideActionFlags sideFlags)
+        private static IEnumerable<RushAction> getActionsFromFlags(RushActionFlags flags)
```

Comment:
Likewise for this:
```cs
private static IEnumerable<RushAction> getActionsFromFlags(RushActionFlags flags)
{
    RushActionFlags flag;
    int action = 0;
    while ((flag = (RushActionFlags)(1 << action++)) <= RushActionFlags.LastAction)
    {
        if (flags.HasFlagFast(flag))
            yield return (RushAction)flag;
    }
}
```

If you add to `RushActionFlags` (please do this anyway):

```cs
LastAction = GroundQuaternary
```